drivers/gpio: ite: Validate num_pins to prevent untrusted loop bound

GTLin08 opened the pull request:
Added a assert to limit num_pins to a maximum allowable value to prevent potential out-of-bounds access or infinite loops.

Fixes: #84805

GTLin08 committed:
drivers/gpio: ite: Modify the format to comply with check_compliance.py

Modify the format to comply with check_compliance.py.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

henrikbrixandersen commented on drivers/gpio/gpio_ite_it8xxx2_v2.c:
```diff
@@ -380,6 +382,8 @@ static void gpio_ite_isr(const void *arg)
 	uint8_t num_pins = gpio_config->num_pins;
 	uint8_t pin;
 
+	__ASSERT(num_pins <= ITE_GPIO_MAX_PINS, "The maximum number of pins per port is 8.");
```
I'd suggest converting this to a `BUILD_ASSERT()` where this is retrieved from the devicetree (`.num_pins = DT_INST_PROP(inst, ngpios),` further down this file).

GTLin08 replied:
Done.

keith-zephyr reviewed (commented):
Can you split the formatting change and the build assert change into 2 commits?

GTLin08 committed:
drivers/gpio: ite: Validate num_pins to prevent untrusted loop bound

Added a build assert to limit num_pins to a maximum allowable value to
prevent potential out-of-bounds access or infinite loops.

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>
